### PR TITLE
Parameterize Time.sleep() with a TimeUnits parameter

### PR DESCRIPTION
**Pull request by nspark**
Updated the existing Time.sleep() function to take a TimeUnits parameter, the default of which is TimeUnits.seconds to maintain compatibility with existing calls. Includes:
 * Supporting runtime changes for Qthreads, FIFO, and MassiveThreads
 * A (rather lenient) test to test/modules/standard/Time, since the OS scheduler can greatly "over sleep" for small nanosleep values

Request review by @gbtitus

### CODE REVIEW

**Comment by nspark** [comment]
I should have added that the local tests in test/modules/standard/Time pass on my machine (linux64) for Qthreads, FIFO, and MassiveThreads.

**Comment by gbtitus** [comment]
One quick bit of feedback -- our automagic PR smoke test failed, apparently due to the lack of a decl for floor() in tasks-qthreads.c.  I think if you set CHPL_DEVELOPER=1 at runtime build time you'll see the error.

**Comment by nspark** [comment]
Thanks. I saw the Travis CI failures and just pushed (what should be) the one-line fix.

File: runtime/include/chpl-tasks.h [hunk 1]
```diff
@@ -193,7 +193,7 @@ void chpl_task_yield(void);
 //
 // Suspend.
 //
-void chpl_task_sleep(int);
+void chpl_task_sleep(int, double);
```
Comment (gbtitus) [inline runtime/include/chpl-tasks.h]: In the interest of simplicity I'm inclined to say let's instead have the runtime chpl_task_sleep() continue to take just a single argument specifying the number of seconds to sleep, but make it a double so that it can express fractions of a second also. This would simplify things by giving us just one copy (in Time.chpl) of the code that converts some number of units into seconds.

File: runtime/src/tasks/fifo/tasks-fifo.c [hunk 2]
```diff
@@ -881,8 +881,11 @@ void chpl_task_yield(void) {
 }
 
 
-void chpl_task_sleep(int secs) {
-  sleep(secs);
+void chpl_task_sleep(int time, double scale) {
+  struct timespec delay;
+  delay.tv_sec = (time_t)(scale*time);
+  delay.tv_nsec = (long)(1e3*(scale*time - floor(scale*time)));
```
Comment (gbtitus) [inline runtime/src/tasks/fifo/tasks-fifo.c]: Seems like we want 1e9 here instead of 1e3? (tv_nsec is nanoseconds)

Comment (nspark) [inline runtime/src/tasks/fifo/tasks-fifo.c]: Oops, that's embarrassing. No wonder my Qthreads tests worked fine, but FIFO and MassiveThreads always seemed a bit off.

File: runtime/include/chpl-tasks.h (hunk 1 above)
Comment (nspark) [inline runtime/include/chpl-tasks.h]: I agree and will update this branch accordingly.

**Comment by gbtitus** [comment]
Looks good! I'll do the corresponding changes in the Cray runtime and we'll be ready to merge.

**Comment by gbtitus** [comment]
Apologies for not catching this before. You may want to assert() in sleep() that t is non-negative.  This didn't occur to me until I consulted the man page for floor() just now. After finding that it truncates toward negative infinity I decided trunc() (which truncates toward zero) might be more appropriate, and then finally the desirability of checking for a negative sleep() popped into my head.

**Comment by nspark** [comment]
I'm inclined to make `sleep()` return immediately if the argument is non-positive rather than halt. Halting would terminate the application, but returning immediately seems more in the spirit of what is requested. We could print a runtime warning to `stderr` saying as much. Thoughts?

**Comment by gbtitus** [comment]
Sure, returning immediately sounds fine.

**Comment by nspark** [comment]
I was going to add the test for negative t in `sleep` when I noticed that it takes a `uint` argument, so it cannot be negative. Since `Timer.elapsed()` and `getCurrentTime()` both return `real` values, it seems most consistent to update the argument type for `sleep()` to `real` and add the warning on negative values.

**Comment by gbtitus** [comment]
Yes, making sleep() take a real seems better to me as well.
